eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py

```diff
@@ -0,0 +1,378 @@
+import argparse
+import glob
+import os
+import psycopg2
+from collections import defaultdict
+from datetime import datetime
+
+from ebi_eva_common_pyutils.logger import logging_config
+from ebi_eva_common_pyutils.mongodb import MongoDatabase
+from ebi_eva_common_pyutils.config_utils import get_pg_metadata_uri_for_eva_profile
+from ebi_eva_common_pyutils.pg_utils import execute_query, get_all_results_for_query
+from gather_per_species_clustering_counts import assembly_table_name
+
+
+logger = logging_config.get_logger(__name__)
+logging_config.add_stdout_handler()
+
+
+def gather_count_from_mongo(clustering_dir, mongo_source, private_config_xml_file, release_version):
+    # Assume the directory structure:
+    # clustering_dir --> <scientific_name_taxonomy_id> --> <assembly_accession> --> cluster_<date>.log_dict
+
+    all_log_pattern = os.path.join(clustering_dir, '*', 'GCA_*', 'cluster_*.log')
+    all_log_files = glob.glob(all_log_pattern)
+    ranges_per_assembly = get_assembly_info_and_date_ranges(all_log_files)
+    metrics_per_assembly = get_metrics_per_assembly(mongo_source, ranges_per_assembly)
+    insert_counts_in_db(private_config_xml_file, metrics_per_assembly, ranges_per_assembly, release_version)
+
+
+def get_assembly_info_and_date_ranges(all_log_files):
+    """
+    Parse all the log files and retrieve assembly basic information (taxonomy, scientific name) and the date ranges
+    where all jobs and steps were run during the clustering process
+    """
+    ranges_per_assembly = defaultdict(dict)
+    for log_file in all_log_files:
+        logger.info('Parse log_dict file: ' + log_file)
+        scientific_name, taxid, assembly_accession, log_date = parse_log_file_path(log_file)
+        log_metric_date_range = parse_one_log(log_file)
+
+        if assembly_accession not in ranges_per_assembly:
+            ranges_per_assembly[assembly_accession] = defaultdict(dict)
+            ranges_per_assembly[assembly_accession]['metrics'] = defaultdict(dict)
+
+        # assembly info
+        ranges_per_assembly[assembly_accession]['taxid'] = taxid
+        ranges_per_assembly[assembly_accession]['scientific_name'] = scientific_name
+
+        # new_remapped_current_rs
+        if 'CLUSTERING_CLUSTERED_VARIANTS_FROM_MONGO_STEP' in log_metric_date_range:
```

Comment:
I think we need to add support for STUDY_CLUSTERING_STEP as well

Reply:
[Addressed](https://github.com/EBIvariation/eva-accession/pull/372/commits/159a8552339a1b45db83eada12660c068aeca79b)